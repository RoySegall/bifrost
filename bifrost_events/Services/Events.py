from bifrost.src.ioc.Base import ServiceBase
from bifrost_events.models import Accommodation, Lunch, Meeting, GeneralEvent, \
    Flight, MeetingConjunction, PickingCar


class Service(ServiceBase):

    def info(self):
        return {
            'name': 'Accommodation service',
            'description': 'Manage accommodations',
        }

    def create_accommodation(self, **kwargs):
        """
        Creating accommodation record in the DB.

        :param kwargs: Arguments of the accommodation model.

        :return: The accommodation service.
        """
        return Accommodation.objects.create(**kwargs)

    def create_lunch(self, **kwargs):
        """
        Creating a lunch event.

        :param kwargs: Arguments for the event.

        :return: The event object.
        """
        return Lunch.objects.create(**kwargs)

    def create_meeting(self, **kwargs):
        """
        Creating a meeting event.

        :param kwargs: Arguments for the event.

        :return: The meeting object.
        """
        return Meeting.objects.create(**kwargs)

    def create_general_event(self, **kwargs):
        """
        Creating a general event.

        :param kwargs: Arguments for the event.

        :return: The meeting object.
        """
        return GeneralEvent.objects.create(**kwargs)

    def create_flight_event(self, **kwargs):
        """
        Creating a flight service.

        :param kwargs: Arguments to pass to the flight model.

        :return: A flight service.
        """
        return Flight.objects.create(**kwargs)

    def create_meeting_conjunction(self, members, **kwargs):
        """
        Create the meeting conjunction object.

        :return:
        """
        meeting_conjunction = MeetingConjunction.objects.create(**kwargs)

        # Adding the members to the meeting conjunction.
        for member in members:
            meeting_conjunction.members.add(member)

        return meeting_conjunction

    def create_picking_car(self, **kwargs):
        """
        Create the picking car event.

        :param kwargs: Arguments to pass to the picking car create object.

        :return: A newly created picking car event.
        """
        return PickingCar.objects.create(**kwargs)
