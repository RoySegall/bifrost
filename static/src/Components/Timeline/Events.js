import React from 'react';
import {X, Car, Hotel, Plane, Friends} from '../../Fonts'
import {
  dateFormat,
  dateFormatOnlyHour,
  dateFormatWithHour
} from '../../Services/consts'
import * as moment from 'moment';

export default class Events extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      events: props['events'],
      activeEvent: null,
      borderColor: {
        accommodationSet: 'pink',
        flightSet: 'orange',
        meetingconjunctionSet: 'blue',
        pickingcarSet: 'green'
      },
    };
  }

  /**
   * Setting which event to display in the vent view. In case we already
   * clicked on the same event the selected event will be deselected.
   *
   * @param selectedEvent
   *  The event we clicked.
   */
  setEventView(selectedEvent) {
    if (this.state.activeEvent) {
      if (
        this.state.activeEvent['day'] === selectedEvent['day'] &&
        this.state.activeEvent['key'] === selectedEvent['key']
      ) {
        this.removeEventView();
        return;
      }
    }

    this.setState({activeEvent: selectedEvent})
  }

  /**
   * Setting the selected event to null which will remove the event view.
   */
  removeEventView() {
    this.setState({activeEvent: null})
  }

  /**
   * This will refactor the end date by given laws: if the end date is in the
   * same day as the starting date only the hours will be displayed. If the
   * end date is in another day then we will display the full date.
   *
   * @param startDate
   *  The starting date of the event.
   * @param endDate
   *  The ending date.
   * @returns {string}
   *  The end date by laws.
   */
  refactorEndDate(startDate, endDate) {
    const startingFrom = moment.unix(startDate).utc();
    const endingFromUnix = moment.unix(endDate).utc();

    if (endingFromUnix.diff(startingFrom, "days") > 0) {
      return endingFromUnix.format(dateFormatWithHour);
    }

    return endingFromUnix.format(dateFormatOnlyHour);
  }

  List(days) {
    return <div>
      <div>
        {
          Object.keys(days).map(day => {
            const events = days[day]['events'].map((event, key) => {
              return this.singleEvent(event, day, key);
            });

            return <>
              {this.dayHead(days[day]['label'])}
              {events}
            </>
          })
        }
      </div>
    </div>
  }

  EventView() {
    if (this.state.activeEvent === null) {
      return <></>;
    }

    const activeEvent = this.state.activeEvent;
    const event = this.state.events[activeEvent['day']]['events'][activeEvent['key']];

    const onClick = (event) => {
      event.preventDefault();
      this.removeEventView();
    };

    const rightBorder = `border-r-2 border-dashed border-${this.state.borderColor[event['type']]}-500`;

    return <div className={`bg-gray-1000 pl-2 ml-2 ${rightBorder}`}>

      <div className={"event-view-wrapper"}>

        <div className="flex flex-row justify-between ml-2 mr-3 mt-2">
          <span>
            <h3 className="text-3xl font-bold">{event['title']}</h3>
          </span>
          <span>
            <a className="text-2xl text-red-600"
               href="#"
               onClick={(event) => onClick(event)}>
              {X()}
            </a>
          </span>
        </div>

        <div className="flex flex-row flex-shrink-0 p-3">
          <img className="border-2 border-red-400 border-solid rounded-lg mt-2"
               src={process.env.PUBLIC_URL + '/mps.png'}/>
        </div>

        <div className="pb-3">
          <span>
              Location: <b>{event['location']['title']}</b>,
              at <b>{moment.unix(event['startingDate']).utc().format(dateFormat)}</b>,&nbsp;
          </span>

          <span>
              Starts at <b>{moment.unix(event['endingDate']).utc().format(dateFormatOnlyHour)}</b> and
              ends at: <b>{this.refactorEndDate(event['startingDate'], event['endingDate'])}</b>
          </span>
        </div>

        <p className="mr-2">
          Lorem Ipsum is simply dummy text of the printing and
          typesetting industry. Lorem Ipsum has been the
          industry's standard dummy text ever since the 1500s,
          when an unknown printer took a galley of type and
          scrambled it to make a type specimen book. It has
          survived not only five centuries, but also the leap
          into electronic typesetting, remaining essentially
          unchanged. It was popularised in the 1960s with the
          release of Letraset sheets containing Lorem Ipsum
          passages, and more recently with desktop publishing
          software like Aldus PageMaker including versions of
          Lorem Ipsum.
        </p>
      </div>
    </div>
  }

  dayHead(title) {
    return <>
      <h3 className="text-3xl m-0 m-auto pb-3 text-center">{title}</h3>
      <hr className="w-1/2 m-0 m-auto p-2 border-purple-400 border-t-2 border-dashed"/>
    </>
  }

  singleEvent(event, day, key) {
    const icons = {
      accommodationSet: Hotel(),
      flightSet: Plane(),
      meetingconjunctionSet: Friends(),
      pickingcarSet: Car(),
    };

    const icon = icons[event['type']];

    const onClick = (event, day, key) => {
      event.preventDefault();
      this.setEventView({day, key});
    };

    const leftBorder = `border-l-2 border-dashed border-${this.state.borderColor[event['type']]}-500`;
    let rightBorder = `border-r-2 border-dashed border-${this.state.borderColor[event['type']]}-500`;

    if (this.state.activeEvent !== null) {
      rightBorder = '';
    }

    return <>
      <div key={key} className={`${leftBorder} ${rightBorder} bg-gray-1000 p-2 mb-4`}>
        <div className="flex flex-row justify-between pl-2 pr-2">
          <span className={`text-2xl text-${this.state.borderColor[event['type']]}-500 font-bold`}>{icon}</span>
          <span>{moment.unix(event['startingDate']).utc().format(dateFormatOnlyHour)}</span>
        </div>

        <hr className="bg-orange-500 mt-2 mb-1"/>

        <div className="flex flex-row justify-between pt-3 pl-2 pr-2">
          <a href='#' className="font-bold text-lg" onClick={(event) => onClick(event, day, key)}>
            {event['title']}
          </a>
          <span className="text-lg">Ends at: {this.refactorEndDate(event['startingDate'], event['endingDate'])}</span>
        </div>

        <p className="pt-3 pl-2 pr-2">
          TBD
        </p>
      </div>
    </>
  }

  render() {
    const cols = this.state.activeEvent !== null ? 'grid-cols-2' : 'grid-cols-1';
    return <>
      <div className={"mt-6 grid " + cols}>
        {this.List(this.state.events)}
        {this.EventView()}
      </div>
    </>
  }
}
