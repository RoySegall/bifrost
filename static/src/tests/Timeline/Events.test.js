import React from 'react';
import renderer from 'react-test-renderer';
import Events from '../../Components/Timeline/Events';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const dummyEvents = JSON.parse('{"17-11-2019":{"timestamp":1573941600,"events":[{"id":"1","title":"Flight to london","origin":"Tel aviv","destination":"London","startingDate":1573970400,"endingDate":1573984800,"location":{"id":"1","title":"jfk airplain","address":"ff"},"connectionFlight":null,"type":"flightSet"},{"id":"1","title":"Hotel Gibson","location":{"id":"2","title":"Gibson hotel","address":"Gibson hotel, Ireland"},"startingDate":1574006400,"endingDate":1574330400,"hotelName":"Gisbon","room":"312","type":"accommodationSet"}],"label":"17-11-2019, Sunday"},"18-11-2019":{"timestamp":1574028000,"events":[{"id":"2","title":"Demo at clinic in London","location":{"id":"1","title":"jfk airplain","address":"ff"},"startingDate":1574064000,"endingDate":1574067600,"hotelName":"foo","room":"123","type":"accommodationSet"},{"id":"3","title":"Lunch with John","location":{"id":"2","title":"Gibson hotel","address":"Gibson hotel, Ireland"},"startingDate":1574071200,"endingDate":1574076600,"hotelName":"asdasd","room":"221","type":"accommodationSet"},{"title":"Picking car","id":"1","startingDate":1574078400,"endingDate":1574078400,"location":{"id":"2","title":"Gibson hotel","address":"Gibson hotel, Ireland"},"type":"pickingcarSet"},{"id":"2","title":"Picking employees","startingDate":1574096400,"endingDate":1574096400,"members":[{"id":"1"},{"id":"2"}],"location":{"id":"1","title":"jfk airplain","address":"ff"},"type":"meetingconjunctionSet"}],"label":"18-11-2019, Monday"},"19-11-2019":{"timestamp":1574114400,"events":[{"id":"4","title":"Demo at a clinic","location":{"id":"1","title":"jfk airplain","address":"ff"},"startingDate":1574143200,"endingDate":1574148600,"hotelName":"asdas","room":"213","type":"accommodationSet"},{"id":"5","title":"Driving to stonehenge","location":{"id":"1","title":"jfk airplain","address":"ff"},"startingDate":1574150400,"endingDate":1574150400,"hotelName":"33","room":"asdd","type":"accommodationSet"},{"id":"6","title":"Luch","location":{"id":"1","title":"jfk airplain","address":"ff"},"startingDate":1574164800,"endingDate":1574170200,"hotelName":"saasd","room":"asdasd","type":"accommodationSet"},{"id":"7","title":"Driving to manchenter","location":{"id":"2","title":"Gibson hotel","address":"Gibson hotel, Ireland"},"startingDate":1574172000,"endingDate":1574193600,"hotelName":"23213","room":"123123","type":"accommodationSet"}],"label":"19-11-2019, Tuesday"},"20-11-2019":{"timestamp":1574200800,"events":[{"id":"8","title":"demo at a clinic","location":{"id":"1","title":"jfk airplain","address":"ff"},"startingDate":1574236800,"endingDate":1574251200,"hotelName":"21321","room":"213213","type":"accommodationSet"},{"id":"9","title":"Back to the hotel","location":{"id":"2","title":"Gibson hotel","address":"Gibson hotel, Ireland"},"startingDate":1574254800,"endingDate":1574254800,"hotelName":"foo","room":"foo","type":"accommodationSet"}],"label":"20-11-2019, Wednesday"}}');
const filterCallbackMock = jest.fn();

test("Testing the setEventView function", () => {
    const component = mount(<Events events={dummyEvents} selectedEventType={'all'}/>);
    const spy = jest.spyOn(component.instance(), 'setEventView');

    expect(component.state('activeEvent')).toBe(null);
    expect(component.exists('.event-view')).toBe(false);

    // Click on the first event and make sure the state has changed.
    component.find('.title.event-flightSet-id-1').simulate('click');
    expect(component.state('activeEvent')).toStrictEqual({
        "day": "17-11-2019",
        "key": 0
    });

    // Making sure the event has been triggered.
    expect(spy).toHaveBeenCalled();
    expect(component.exists('.event-view')).toBe(true);

    // Click on the second event and make sure the state has changed to that event.
    component.find('.title.event-accommodationSet-id-1').simulate('click');
    expect(component.state('activeEvent')).toStrictEqual({
        "day": "17-11-2019",
        "key": 1
    });

    // Click on event which is already selected and make sure the state changed.
    component.find('.title.event-accommodationSet-id-1').simulate('click');
    expect(component.state('activeEvent')).toBe(null);
    expect(component.exists('.event-view')).toBe(false);
});

test("Testing the removeEventView event", () => {
    const component = mount(<Events events={dummyEvents} selectedEventType={'all'}/>);
    const spy = jest.spyOn(component.instance(), 'removeEventView');

    // Set an event to be viewed.
    component.find('.title.event-accommodationSet-id-1').simulate('click');

    // Making sure the event view exists.
    expect(component.exists('.event-view')).toBe(true);

    // Simulating the event close and making sure the event was triggered.
    component.find('.close-event').simulate('click');
    expect(spy).toHaveBeenCalled();

    // Making sure the event view does not exists.
    expect(component.exists('.event-view')).toBe(false);
});

test("Testing the refactorEndDate method", () => {
    const component = mount(<Events events={dummyEvents} selectedEventType={'all'}/>);
    const instance = component.instance();

    // Making sure that if the event is in the same day we won't display the
    // full date.
    expect(instance.refactorEndDate(1573970400, 1573970400 + 3600)).toBe("07:00");

    // Making sure that when the event is another day we will display the full
    // day.
    expect(instance.refactorEndDate(1573970400, 1573970400 + (84600 * 2))).toBe("19-11-2019 05:00");
});

test('Testing the filter by event type', () => {
    // Testing all the events exists.
    let component = mount(<Events events={dummyEvents} selectedEventType={'all'}/>);

    // Set an event to be viewed.
    expect(component.exists('.title.event-flightSet-id-1')).toBe(true);
    expect(component.exists('.title.event-accommodationSet-id-1')).toBe(true);

    // Testing only flight type exists.
    component = mount(<Events events={dummyEvents} selectedEventType={'flightSet'}/>);
    expect(component.exists('.title.event-flightSet-id-1')).toBe(true);
    expect(component.exists('.title.event-accommodationSet-id-1')).toBe(false);

    // Testing only the accommodation events will be displayed.
    component = mount(<Events events={dummyEvents} selectedEventType={'accommodationSet'}/>);
    expect(component.exists('.title.event-flightSet-id-1')).toBe(false);
    expect(component.exists('.title.event-accommodationSet-id-1')).toBe(true);
});

test("Testing the eventHead, eventBody, EventView function with snapshot", () => {
    const EventComponent = renderer.create(<Events events={dummyEvents} selectedEventType={'all'}/>);
    expect(EventComponent.toJSON()).toMatchSnapshot();
});
