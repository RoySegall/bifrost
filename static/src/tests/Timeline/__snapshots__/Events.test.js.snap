// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the eventHead, eventBody, EventView function with snapshot 1`] = `
<div
  className="mt-6 grid grid-cols-1"
>
  <div>
    <div>
      <h3
        className="text-3xl m-0 m-auto pb-3 text-center"
      >
        17-11-2019, Sunday
      </h3>
      <hr
        className="w-1/2 m-0 m-auto p-2 border-purple-400 border-t-2 border-dashed"
      />
      <div
        className="border-l-2 border-dashed border-orange-500 border-r-2 border-dashed border-orange-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-orange-500 font-bold"
          >
            <i
              className="fal fa-plane"
            />
          </span>
          <span>
            06:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Flight to london
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            10:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            16:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Hotel Gibson
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            21-11-2019 10:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <h3
        className="text-3xl m-0 m-auto pb-3 text-center"
      >
        18-11-2019, Monday
      </h3>
      <hr
        className="w-1/2 m-0 m-auto p-2 border-purple-400 border-t-2 border-dashed"
      />
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            08:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Demo at clinic in London
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            09:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            10:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Lunch with John
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            11:30
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-green-500 border-r-2 border-dashed border-green-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-green-500 font-bold"
          >
            <i
              className="fal fa-car"
            />
          </span>
          <span>
            12:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Picking car
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            12:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-blue-500 border-r-2 border-dashed border-blue-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-blue-500 font-bold"
          >
            <i
              className="fal fa-user-friends"
            />
          </span>
          <span>
            17:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Picking employees
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            17:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <h3
        className="text-3xl m-0 m-auto pb-3 text-center"
      >
        19-11-2019, Tuesday
      </h3>
      <hr
        className="w-1/2 m-0 m-auto p-2 border-purple-400 border-t-2 border-dashed"
      />
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            06:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Demo at a clinic
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            07:30
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            08:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Driving to stonehenge
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            08:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            12:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Luch
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            13:30
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            14:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Driving to manchenter
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            20:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <h3
        className="text-3xl m-0 m-auto pb-3 text-center"
      >
        20-11-2019, Wednesday
      </h3>
      <hr
        className="w-1/2 m-0 m-auto p-2 border-purple-400 border-t-2 border-dashed"
      />
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            08:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            demo at a clinic
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            12:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
      <div
        className="border-l-2 border-dashed border-pink-500 border-r-2 border-dashed border-pink-500 bg-gray-1000 p-2 mb-4"
      >
        <div
          className="flex flex-row justify-between pl-2 pr-2"
        >
          <span
            className="text-2xl text-pink-500 font-bold"
          >
            <i
              className="fal fa-hotel"
            />
          </span>
          <span>
            13:00
          </span>
        </div>
        <hr
          className="bg-orange-500 mt-2 mb-1"
        />
        <div
          className="flex flex-row justify-between pt-3 pl-2 pr-2"
        >
          <a
            className="font-bold text-lg"
            href="#"
            onClick={[Function]}
          >
            Back to the hotel
          </a>
          <span
            className="text-lg"
          >
            Ends at: 
            13:00
          </span>
        </div>
        <p
          className="pt-3 pl-2 pr-2"
        >
          TBD
        </p>
      </div>
    </div>
  </div>
</div>
`;
